src/botkit-witai.js

```diff
@@ -48,8 +47,8 @@ module.exports = function (options) {
     if (message.entities && Object.keys(message.entities).length) {
       for (var i = 0; i < Object.keys(message.entities).length; i++) {
         for (var t = 0; t < patterns.length; t++) {
-          if (Object.keys(message.entities)[i] == patterns[t]){
-            for (var j = 0; j < message.entities[Object.keys(message.entities)[i]].length; j++){
+          for (var j = 0; j < message.entities[Object.keys(message.entities)[i]].length; j++){
+            if (message.entitites[Object.keys(message.entities)[i]][j].value == patterns[t]){
```

Comment:
Fix you typo:
`if (message.entities[Object.keys(message.entities)[i]][j].value == patterns[t]){`